aten/src/ATen/native/cuda/Reduce.cuh

```diff
@@ -733,7 +732,7 @@ struct ReduceOp {
     at::detail::Array<arg_t, output_vec_size>,
     typename std::enable_if<!can_acc>::type* = nullptr
   ) const {
-    assert(false); // can't use AT_ASSERT in Cuda.
+    CUDA_KERNEL_ASSERT(false); // can't use AT_CUDA_KERNEL_ASSERT in Cuda.
```

Comment:
```suggestion
    CUDA_KERNEL_ASSERT(false);
```
Don't need the comment now.  AT_ASSERT is deprecated in favor of TORCH_INTERNAL_ASSERT or TORCH_CHECK, but since this is device code here CUDA_KERNEL_ASSERT is appropriate.

Reply:
Haha, thanks. it was a weird replacement.